gridcomps/Cap/MAPL_CapGridComp.F90

```diff
@@ -1569,6 +1678,222 @@ subroutine rewind_clock(this, time, rc)
     _RETURN(_SUCCESS)
   end subroutine rewind_clock
 
+  subroutine step_reverse(this, unusable, phase, first, rc)
+    class(MAPL_CapGridComp), intent(inout) :: this
+    class(KeywordEnforcer), optional, intent(in   ) :: unusable
+    integer, optional, intent(it) :: phase
+    logical, intent(in)  :: first
+    integer, intent(out) :: rc
+
+    integer :: status, phase_
+
+
+    type(ESMF_Time) :: currTime
+
+    _UNUSED_DUMMY(unusable)
+    phase_ = 1
+    if (present(phase)) phase_ = phase
+
+    call ESMF_GridCompGet(this%gc, vm = this%vm)
+    ! Run the ExtData Component
+    ! --------------------------
+    if (phase_ == 1) then
+       call first_phase(first, rc=status)
+       _VERIFY(status)
+
+    endif ! phase_ == 1
+
+
+    ! Run the Gridded Component
+    ! --------------------------
+    call ESMF_GridCompRun(this%gcs(this%root_id), importstate = this%child_imports(this%root_id), &
+         exportstate = this%child_exports(this%root_id), &
+         clock = this%clock, userrc = status)
+    _VERIFY(status)
+
+    ! Synchronize for Next TimeStep
+    ! -----------------------------
+
+    call ESMF_VMBarrier(this%vm, rc = status)
+    _VERIFY(STATUS)
+
+    ! Call History Run for Output
+    ! ---------------------------
+
+    call ESMF_GridCompRun(this%gcs(this%history_id), importstate=this%child_imports(this%history_id), &
+         exportstate = this%child_exports(this%history_id), &
+         clock = this%clock_hist, userrc = status)
+    _VERIFY(status)
```

Comment:
Need to check `rc` _and_ `userrc`.   Probably need an additional local variable for the latter.  We usually spell it `userRC`